### DISCUSSION TITLE
Unable to ping from MAC although connection is showing up

### DISCUSSION THREAD

**Discussion by salmanmalik-emb** [strongswan/strongswan#1772] [Q&A]
I have setup of strongswan on both server and client. I am using local proxy to send and received udp packets from/to port 4500. Same setup is working with ubuntu OS and not working with MAC (although configs and routes are exactly same)

It shows following results captured through wireshark, tcpdump

Ping from Client to Server
1. Packet reaches to server
2. Packet received to ipsec port 4500 on server
3. Packet sent back to the client
4. Packet received on client
5. Packet received to ipsec port 4500 on client
5. Ping timed out

What could be the possible issue that on mac, packets are not processed or forwarded further by 4500 ? How can I check strongswan logs to view what is going on with the packets received on 4500 service on mac?

**Comment by tobiasbrunner** [comment]
See [the docs](https://docs.strongswan.org/docs/5.9/config/logging.html) for log settings.

**Comment by salmanmalik-emb** [comment]
@tobiasbrunner I have enabled logs mentioned in this document, but traffic logs are not there. From where I can see the traffic being processed by IPSec ?

**Reply by tobiasbrunner** [reply]
Obviously not if the traffic is handled by the kernel. But you might see something else in the log.

**Reply by salmanmalik-emb** [reply]
I can see this in logs 

`unable to query policy 100.64.0.196/32 === 100.64.0.204/32 out: kernel reports no use time`

**Reply by tobiasbrunner** [reply]
That shouldn't be an issue.

**Reply by salmanmalik-emb** [reply]
Also I can see following as inbound packets with no SA 

How can I debug why SA is not available or mismatched ?

sh-3.2# netstat -ss -p ipsec
ipsec:
	329 inbound packets processed successfully
	1538 inbound packets with no SA available
	3 inbound packets failed on ESP replay check
	ESP input histogram:
		rijndael-cbc: 329
	2234 outbound packets processed successfully
	30 outbound packets with no SA available
	ESP output histogram:
		rijndael-cbc: 2234

**Reply by salmanmalik-emb** [reply]
Also I am repeatedly seeing following in logs

`received PF_KEY message with unexpected sequence number, was 0 expected 49`

**Reply by tobiasbrunner** [reply]
> 1538 inbound packets with no SA available

That doesn't sound too good. But no idea what the reason could be exactly.





> Also I am repeatedly seeing following in logs
> 
> `received PF_KEY message with unexpected sequence number, was 0 expected 49`

I don't think that's an issue, just some macOS quirk.

**Reply by salmanmalik-emb** [reply]
@tobiasbrunner Can you please suggest different ways or tools to debug further so I can get to the root cause of this issue?

**Comment by salmanmalik-emb** [comment]
@tobiasbrunner  Can you or someone you know can help us debug this issue on Mac, we can have a paid gig for it.

**Reply by Kentzo** [reply]
Try wireshark: https://docs.strongswan.org/docs/5.9/install/trafficDumps.html

I don’t use strongswan on macOS but I use its native client to connect to strongswan and it works well.

**Comment by salmanmalik-emb** [comment]
@tobiasbrunner found the issue, we are using local proxy for our specify case and MacOS doesnt distinguish port based SAs and deletes duplicate SA with src and dst IPs are same, which were in our case 127.0.0.1 because of local proxy. We have fixed it by using two localhosts IPs 127.0.0.2 and 127.0.0.1
